src/averageAges.js

```diff
@@ -1,61 +1,69 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+function calculateAverageAge(people, filterFunction) {
+  const filteredPeople
+    = filterFunction ? people.filter(filterFunction) : people;
+
+  const total = filteredPeople.reduce((sum, { born, died }) => {
+    return sum + (died - born);
+  }, 0);
+
+  const result = total / filteredPeople.length;
+
+  return result;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
+function calculateMenAverageAge(people, century) {
+  const filterFunction = (person) => person.sex === 'm';
```

Comment:
**[severity 2]: Inconsistent naming of variables**
            Variable 'filteredFunction' is less descriptive than other variables, use a more descriptive name
            **How to fix:**
            Rename the variable 'filteredFunction' to 'centuryFilterFunction'